core/state/statedb.go

```diff
@@ -1373,64 +1367,91 @@ func (s *StateDB) Commit(block uint64, deleteEmptyObjects bool) (common.Hash, er
 // - Reset transient storage (EIP-1153)
 func (s *StateDB) Prepare(rules params.Rules, sender, coinbase common.Address, dst *common.Address, precompiles []common.Address, list types.AccessList) {
 	if rules.IsBerlin {
-		// Clear out any leftover from previous executions
-		al := newAccessList()
-		s.accessList = al
+		if !rules.IsPrague {
+			s.accessList = newAccessList()
+		} else {
+			s.accessList = s.NewAccessWitness()
+		}
+		al := s.accessList
 
-		al.AddAddress(sender)
+		al.AddAddress(sender, ALAllItems, false)
```

Comment:
Suggestion: use the defined enums instead of bools? (Same with lines below)

Reply:
yeah, vscode autofilled it for me :(